server/docker-compose.yml

```
@@ -1,95 +1,143 @@
+# Updated based on v1.2.0 `prefect server config` command
 version: "3.7"
 
 services:
 
   postgres:
-    image: "postgres:11"
+    command:
+      - postgres
+      - -c
+      - max_connections=150
     environment:
       POSTGRES_USER: ${POSTGRES_USER}
       POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
       POSTGRES_DB: ${POSTGRES_DB}
     volumes:
-      - postgres:/var/lib/postgresql/data
+        - postgres:/var/lib/postgresql/data
+    healthcheck:
+      interval: 10s
+      retries: 60
+      start_period: 2s
+      test: pg_isready -q -d $${POSTGRES_DB} -U $${POSTGRES_USER} || exit 1
+      timeout: 2s
+    image: postgres:11
     networks:
-      - prefect-server
-    restart: "always"
-    command:
-      - "postgres"
-      - "-c"
-      - "max_connections=150"
+      prefect-server: null
+    restart: always
     
   hasura:
-    image: "hasura/graphql-engine:v1.3.3"
-    command: "graphql-engine serve"
+    command: graphql-engine serve
+    depends_on:
+      postgres:
+        condition: service_started
     environment:
-      HASURA_GRAPHQL_DATABASE_URL: ${DB_CONNECTION_URL}
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+      HASURA_GRAPHQL_DATABASE_URL: postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}
       HASURA_GRAPHQL_ENABLE_CONSOLE: "true"
-      HASURA_GRAPHQL_SERVER_PORT: "3000"
+      HASURA_GRAPHQL_LOG_LEVEL: warn
       HASURA_GRAPHQL_QUERY_PLAN_CACHE_SIZE: 100
-      HASURA_GRAPHQL_LOG_LEVEL: "warn"
+      HASURA_GRAPHQL_SERVER_PORT: '3000'
+      HASURA_GRAPHQL_V1_BOOLEAN_NULL_COLLAPSE: "true"
+    healthcheck:
+      interval: 10s
+      retries: 60
+      start_period: 1s
+      test: wget -O - http://hasura:$${HASURA_GRAPHQL_SERVER_PORT}/healthz &>/dev/null
+        || exit 1
+      timeout: 2s
+    image: hasura/graphql-engine:v2.1.1
+    init: true
     networks:
-      - prefect-server
-    restart: "always"
-    depends_on:
-      - postgres
+      prefect-server: null
+    restart: always
+
   
   graphql:
-    image: "prefecthq/server:core-0.15.3"
-    command: bash -c "${PREFECT_SERVER_DB_CMD} && python src/prefect_server/services/graphql/server.py"
+    command: bash -c "prefect-server database upgrade -y && python src/prefect_server/services/graphql/server.py"
+    depends_on:
+      hasura:
+        condition: service_started
     environment:
-      PREFECT_SERVER_DB_CMD: ${PREFECT_SERVER_DB_CMD:-"echo 'DATABASE MIGRATIONS SKIPPED'"}
-      PREFECT_SERVER__DATABASE__CONNECTION_URL: ${DB_CONNECTION_URL}
-      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${PREFECT_SERVER__HASURA__ADMIN_SECRET:-hasura-secret-admin-secret}
+      PREFECT_CORE_VERSION: 1.2.0
+      PREFECT_SERVER_DB_CMD: prefect-server database upgrade -y
+      PREFECT_SERVER__DATABASE__CONNECTION_URL: postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}
+      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${HASURA_GRAPHQL_ADMIN_SECRET}
```

This value is missing in `server/.env`, we could replace it by a default value with :

```suggestion
      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${PREFECT_SERVER__HASURA__ADMIN_SECRET:-hasura-secret-admin-secret}
```